packages/@uppy/tus/src/index.ts

```diff
@@ -298,57 +357,83 @@ export default class Tus extends BasePlugin {
             }
             this.requests.rateLimit(next.value)
           }
-        } else if (status > 400 && status < 500 && status !== 409 && status !== 423) {
+        } else if (
+          status != null &&
+          status > 400 &&
+          status < 500 &&
+          status !== 409 &&
+          status !== 423
+        ) {
           // HTTP 4xx, the server won't send anything, it's doesn't make sense to retry
           // HTTP 409 Conflict (happens if the Upload-Offset header does not match the one on the server)
           // HTTP 423 Locked (happens when a paused download is resumed too quickly)
           return false
-        } else if (typeof navigator !== 'undefined' && navigator.onLine === false) {
+        } else if (
+          typeof navigator !== 'undefined' &&
+          navigator.onLine === false
+        ) {
           // The navigator is offline, let's wait for it to come back online.
           if (!this.requests.isPaused) {
             this.requests.pause()
-            window.addEventListener('online', () => {
-              this.requests.resume()
-            }, { once: true })
+            window.addEventListener(
+              'online',
+              () => {
+                this.requests.resume()
+              },
+              { once: true },
+            )
           }
         }
         queuedRequest.abort()
         queuedRequest = {
           shouldBeRequeued: true,
-          abort () {
+          abort() {
             this.shouldBeRequeued = false
           },
-          done () {
-            throw new Error('Cannot mark a queued request as done: this indicates a bug')
+          done() {
+            throw new Error(
+              'Cannot mark a queued request as done: this indicates a bug',
+            )
           },
-          fn () {
+          fn() {
             throw new Error('Cannot run a queued request: this indicates a bug')
           },
         }
         return true
       }
 
       if (opts.onShouldRetry != null) {
-        uploadOptions.onShouldRetry = (...args) => opts.onShouldRetry(...args, defaultOnShouldRetry)
+        uploadOptions.onShouldRetry = (...args) =>
+          // @ts-expect-error TS does not pick up the != null check and also thinks ...args already
+          // contains the `next` argument we are about to add. It would require complex utility types
+          // to fix but it doesn't matter, the type is correct everywhere, it's just confused about this composition.
+          opts.onShouldRetry(...args, defaultOnShouldRetry)
```

Comment:
i think this can be solved by simply pulling it out to a variable (`const onShouldRetry = opts.onShouldRetry`), then I think the `!= null` check will remove nullish

Reply:
It's mostly the other issue I mentioned in that comment that's unnecessarily complex to solve for no gains. If we're already ignoring the error, might as well keep the source code the same.

Reply:
well we are lying to typescript (and ourselves) by disabling type checks and instead using `as tus.UploadOptions` below. I had a really hard understanding what that code was even doing, until removing the `@ts-expect-error` and `as` keyword which revealed what is actually happening here.